server/src/main/scala/passengersplits/core/ZipUtils.scala

```diff
@@ -1,36 +1,30 @@
 package passengersplits.core
 
-import java.io.Closeable
 import java.nio.charset.StandardCharsets._
 import java.util.zip.{ZipEntry, ZipInputStream}
-import scala.collection.JavaConversions._
+
 import scala.collection.mutable.ArrayBuffer
 
 object ZipUtils {
 
   case class UnzippedFileContent(filename: String, content: String, zipFilename: Option[String] = None)
 
-  def usingZip[R <: Closeable, T](unzippedStream: R)(f: (R) => T) = {
-    try {
-      f(unzippedStream)
-    } finally {
-      unzippedStream.close()
-    }
-  }
-
-  def unzipAllFilesInStream(unzippedStream: ZipInputStream): Stream[UnzippedFileContent] = {
-    unzipAllFilesInStream(unzippedStream, Option(unzippedStream.getNextEntry))
+  def unzipAllFilesInStream(unzippedStream: ZipInputStream, fileNameFilter: String => Boolean): Stream[UnzippedFileContent] = {
+    unzipAllFilesInStream(unzippedStream, Option(unzippedStream.getNextEntry), fileNameFilter)
```

Comment:
does it matter that we are no longer closing it?

Reply:
Good question! I've just refactored the zip stuff and added a close() where appropriate for good practise